source/transactions/transactions.rst

```diff
@@ -770,23 +769,18 @@ Drivers MUST send all commands for a single transaction to the same mongos
 After the driver selects a mongos for the first command within a transaction,
 the driver MUST pin the ClientSession to the selected mongos. Drivers MUST
 send all subsequent commands that are part of the same transaction (excluding
-retries of commitTransaction and abortTransaction) to the same mongos.
+certain retries of commitTransaction and abortTransaction) to the same mongos.
 
 When to unpin
 ^^^^^^^^^^^^^
 
-Drivers MUST unpin a ClientSession when a command within a transaction
-fails with a TransientTransactionError. Transient errors indicate that the
-transaction in question has already been aborted or that the pinned mongos is
+Drivers MUST unpin a ClientSession when a command within a transaction,
+including commitTransaction and abortTransaction, fails with a
+TransientTransactionError. Transient errors indicate that the transaction
```

Comment:
I think I commented on this earlier, but `commitTransaction` would never fail with a TransientTransactionError. Do you need to specify UnknownTransactionCommitResult here?

Reply:
The commitTransaction helper never fails with TransientTransactionError but the individual commands run by the helper can. The spec is meant to mandate that when the initial commitTransaction command attempt fails with a TransientTransactionError, drivers must unpin before running any automatic retry attempt.

Reply:
The behavior is tested here: https://github.com/mongodb/specifications/pull/471/files#diff-55cfe16e56662a302c7ec014fbadadaaR173

Reply:
I updated the spec to say:
```
Additionally, drivers MUST unpin a ClientSession when any individual
commitTransaction command attempt fails with an UnknownTransactionCommitResult
error label. In cases where the UnknownTransactionCommitResult causes an
automatic retry attempt, drivers MUST unpin the ClientSession before performing
server selection for the retry.
```